### PR TITLE
unicore: extract vertical velocity and velocity accuracy

### PR DESCRIPTION
**Pull request by julianoes**
It looks like vertical velocity, and velocity accuracy is not available using NMEA, so we extract them from the Unicore ACRICA message instead.

### CODE REVIEW

File: src/unicore.cpp [hunk 1]
```diff
@@ -179,5 +179,31 @@ bool UnicoreParser::extractHeading()
...
+	unsigned i = 0;
+
+	while (ptr != nullptr) {
+		ptr = strtok(nullptr, ",");
+
+		if (ptr == nullptr) {
+			return false;
+		}
+
+		if (i == 24) {
```
Comment (dagar) [inline src/unicore.cpp:200]: Where are you getting 24?

Comment (dagar) [reply src/unicore.cpp:200]: I'm not sure if I have the right document.

image

Comment (dagar) [reply src/unicore.cpp:200]: Including the speed accuracy is also super valuable.

Comment (julianoes) [reply src/unicore.cpp:200]: It's off by a few due to the header. Basically trial and error to get it right.

Ok, I will include speed accuracy.

Comment (julianoes) [reply src/unicore.cpp]: @dagar I've added speed accuracy, and moved all speed fields over to AGRICA.

**Comment by julianoes** [comment]
Data from a flight test looks sane:

velocities
velocities and stddev

File: src/nmea.cpp [hunk 2]
```diff
@@ -922,13 +926,24 @@ int GPSDriverNMEA::receive(unsigned timeout)
...
+
+					_gps_position->vel_m_s = _unicore_parser.agrica().velocity_m_s;
+					_gps_position->vel_n_m_s = _unicore_parser.agrica().velocity_north_m_s;
+					_gps_position->vel_e_m_s = _unicore_parser.agrica().velocity_east_m_s;
+					_gps_position->vel_d_m_s = -_unicore_parser.agrica().velocity_up_m_s;
+					_gps_position->s_variance_m_s =
+						(_unicore_parser.agrica().stddev_velocity_north_m_s * _unicore_parser.agrica().stddev_velocity_north_m_s +
+						 _unicore_parser.agrica().stddev_velocity_east_m_s * _unicore_parser.agrica().stddev_velocity_east_m_s +
+						 _unicore_parser.agrica().stddev_velocity_up_m_s * _unicore_parser.agrica().stddev_velocity_up_m_s)
+						/ 3.0f;
```
Comment (dagar) [inline src/nmea.cpp:944]: Shouldn't this be square root?

Comment (julianoes) [reply src/nmea.cpp:944]: Not sure, it's called variance...

**Pull request merged**